kornia/filters/bilateral.py

```diff
@@ -48,71 +99,54 @@ def bilateral_blur(
         >>> output.shape
         torch.Size([2, 4, 5, 5])
     """
-    KORNIA_CHECK_IS_TENSOR(input)
-    KORNIA_CHECK_SHAPE(input, ['B', 'C', 'H', 'W'])
-    if isinstance(sigma_color, Tensor):
-        KORNIA_CHECK_SHAPE(sigma_color, ['B'])
-        sigma_color = sigma_color.to(device=input.device, dtype=input.dtype).view(-1, 1, 1, 1, 1)
-
-    kx, ky = _unpack_2d_ks(kernel_size)
-    pad_x, pad_y = _compute_zero_padding(kernel_size)
+    return _bilateral_blur(input, None, kernel_size, sigma_color, sigma_space, border_type, color_distance_type)
 
-    padded = pad(input, (pad_x, pad_x, pad_y, pad_y), mode=border_type)
-    unfolded = padded.unfold(2, ky, 1).unfold(3, kx, 1).flatten(-2)  # (B, C, H, W, K x K)
 
-    diff = unfolded - input.unsqueeze(-1)
-    if color_distance_type == "l1":
-        color_distance_sq = diff.abs().sum(1, keepdim=True).square()
-    elif color_distance_type == "l2":
-        color_distance_sq = diff.square().sum(1, keepdim=True)
-    else:
-        raise ValueError("color_distance_type only acceps l1 or l2")
-    color_kernel = (-0.5 / sigma_color**2 * color_distance_sq).exp()  # (B, 1, H, W, K x K)
-
-    space_kernel = get_gaussian_kernel2d(kernel_size, sigma_space, device=input.device, dtype=input.dtype)
-    space_kernel = space_kernel.view(-1, 1, 1, 1, kx * ky)
-
-    kernel = space_kernel * color_kernel
-    out = (unfolded * kernel).sum(-1) / kernel.sum(-1)
-    return out
-
-
-class BilateralBlur(Module):
-    r"""Blur a tensor using a Bilateral filter.
+def joint_bilateral_blur(
+    input: Tensor,
+    guidance: Tensor,
+    kernel_size: tuple[int, int] | int,
+    sigma_color: float | Tensor,
+    sigma_space: tuple[float, float] | Tensor,
+    border_type: str = 'reflect',
+    color_distance_type: str = "l1",
+) -> Tensor:
+    r"""Blur a tensor using a Joint Bilateral filter.
 
```

Comment:
Add the generated image.

Reply:
What should be the guidance image?

Reply:
Anything related to Lena :)
http://www.lenna.org/

Reply:
I'm using the simba image from `kornia.color` sample image. What do you think?

![image](https://user-images.githubusercontent.com/26946864/222712832-59253545-0eb2-4a1d-a137-762c4cc8632e.png)

Reply:
I like this one, it looks easy to saw at least